salt/metalk8s/addons/dex/deployed/service-configuration.sls

```diff
@@ -0,0 +1,66 @@
+include:
+  - .namespace
+
+{%- set dex_config = salt.metalk8s_kubernetes.get_object(
+        kind='ConfigMap',
+        apiVersion='v1',
+        namespace='metalk8s-auth',
+        name='metalk8s-dex-config'
+  )
+%}
+
+{%- if dex_config is none %}
+
+Create dex-config ConfigMap:
+  metalk8s_kubernetes.object_present:
+    - manifest:
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: metalk8s-dex-config
+          namespace: metalk8s-auth
+        data:
+          config.yaml: |-
+            apiVersion: addons.metalk8s.scality.com
+            kind: DexConfig
+            spec:
+              deployment:
+                replicas: 2
+              localuserstore:
+                enabled: true
+                email: "admin@metalk8s.invalid"
+                hash: "$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W"
+              connectors:
+                enabled: false
+                type: ldap
+                name: LDAP
+                id: ldap
+                config:
+                  host: ldap.metalk8s.invalid:636
+                  startTLS: true
+                  rootCAData: "{{ salt['hashutil.base64_encodefile']('/etc/metalk8s/pki/dex/ca.crt') | replace('\n', '') }}"
+                  bindDN: <user>
+                  bindPW: <pwd>
+                  usernamePrompt: SSO Username
+                  userSearch:
+                    baseDN: <baseDN>
+                    filter: "(&(objectclass=user))"
+                    username: <uid>
+                    idAttr: <uidNumber>
+                    emailAttr: <mail>
+                    nameAttr: <name>
+                  groupSearch:
+                    baseDN: <baseDN>
+                    filter: "(objectClass=group)"
+                    userMatchers:
+                      userAttr: <uid>
+                      groupAttr: <member>
+                    nameAttr: <name>
```

Comment:
All the field with `<blabla>` are mandatory ? 
If no I think we may want to pre-fill field with default value (maybe `null`) so that a user can easily just change mandatory field and set `enabled` to `True` and ... let's go

Reply:
IMHO (see above) those fields shouldn't be here at all. It's super confusing.

Reply:
Yes agree we discussed this directly with @Ebaneck also look like the `enabled` key do not work so basically with default bootstrap deployment you will have an "ldap" button in UI that basically do not work.
Just asked, for first iteration, to just copy `connectors` contains in the dex configuration and not checking field one by one + adding some documentation for ldap